### PR TITLE
Add search option in the toolbar for CityFragment

### LINKED ISSUE #167: Create Android search dialog on Home Page
In the Home Page, when we click on "Search a city" edit text, open Android Search Dialog with the suggestions. In the current UI, the suggestions are shown, but in a simple autocomplete list view only.
The UI should be like below:
aww0s

screenshot_20180701-142644

### PR DESCRIPTION
**Pull request by amy6**
# Description

Added MaterialSearchView to the CityFragment - implements auto-complete & suggestions feature.

searchview

Fixes #167 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

**Additional notes**:
This might require further changes, to make the SearchView EditText appear in the toolbar instead of below it. Ran into some issues with handling the 3rd party library and inflating the SearchView into the Fragment's toolbar. Will add in the fix once I figure it out.

Many thanks to @knightcube for helping me fix the issues.

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 1]
```diff
@@ -47,13 +52,15 @@
 
 public class CityFragment extends Fragment {
 
-    @BindView(R.id.cityname)
-    AutoCompleteTextView cityname;
+//    @BindView(R.id.cityname)
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: Remove this.

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 2]
```diff
@@ -82,19 +89,41 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         mToken = sharedPreferences.getString(USER_TOKEN, null);
 
         mHandler = new Handler(Looper.getMainLooper());
-        cityname.setThreshold(1);
+
+        materialSearchView = view.findViewById(R.id.search_view);
+        materialSearchView.setOnQueryTextListener(new MaterialSearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                //Do some magic
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: Remove this comment.

**Comment by amy6** [comment]
@Swati4star The search appears only in the CityFragment as it is added to the fragment layout file.
I have made the suggested changes and fixed the checkstyle errors - squashed commits into one.
Let me know if there's anything else.

**Comment by Swati4star** [comment]
No, it's perfect. I have merged it.

**Pull request merged**